core/modules/node/node.module

```diff
@@ -1470,8 +1470,15 @@ function node_search_execute($keys = NULL, $conditions = NULL) {
 /**
  * Implements hook_ranking().
  */
-function node_ranking() {
+function node_ranking($keywords = NULL) {
   // Create the ranking array and add the basic ranking options.
+  $words = explode(' ', $keywords);
+  $conditions = array();
+  foreach($words as $word) {
+    $conditions[] = "n.title LIKE '%" . $word . "%'";
+  }
+  $sql_string = implode(' OR ', $conditions);
```

Comment:
One thing about this approach is that **any** word that matches in the title gives a score of `1`. Since the score range is a decimal, I think the more words that match, the greater the score should be, correct?

So ideally, if the query had 3 words but one matched, the score would be `0.333`. If two matched, it would be `0.666`. And if all words match it's `1`.

Reply:
I started there too, but couldn't figure out how to do it with SQL.

Reply:
So this gets complicated, but I think the query would end up looking like:

$score_per_word = 1 / count($keywords);

`SUM(IF((n.title LIKE '%foo%'), $score_per_word, 0), IF((n.title LIKE '%bar%'), $score_per_word, 0), IF((n.title LIKE '%baz%'), $score_per_word, 0))`

Reply:
Resolving this conversation since I don't think this approach is feasible.